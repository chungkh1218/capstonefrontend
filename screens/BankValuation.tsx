import * as React from "react";
import { Component } from "react";
import {
  View,
  TouchableOpacity,
  Text,
  Dimensions,
  StyleSheet,
  FlatList,
  Image,
  ScrollView
} from "react-native";
import { Linking } from "react-native";
import MortgageInput from "../components/mortgage/mortgagecalculator";
import Header from "../components/mortgage/mortgagehead";
import { Button } from "react-native-elements";
interface IBankValuationProps {}
interface IBankValuationState {
  data: Array<{
    name: string;
    url: string;
  }>;
}
export default class BankValuation extends Component<
  IBankValuationProps,
  IBankValuationState
> {
  constructor(props: IBankValuationProps) {
    super(props);
    this.state = {
      data: [
        {
          name: "HSBC",
          url: "../src/img/hsbc_logo.png"
        },
        {
          name: "Hang Seng Bank",
          url: "../src/img/hsb_logo.png"
        },
        {
          name: "DBS",
          url: "../src/img/dbs_logo.png"
        },
        {
          name: "Standard Chartered",
          url: "../src/img/sc_logo.png"
        }
      ]
    };
  }
  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={this.state.data}
          renderItem={({ item }) => (
            <TouchableOpacity onPress={() => Linking.openURL(item.url)}>
              <View style={styles.homeList}>
                <Text style={{ padding: 20, fontSize: 18 }}>{item.name}</Text>
                {/* <Image
                  style={{ width: 80, height: 80, backgroundColor: "red" }}
                  source={{ uri: item.url.toString() }}
                  source={require(item.url.toString())}
                /> */}
              </View>
            </TouchableOpacity>
          )}
          keyExtractor={(item, key) => String(key)}
        />

        <Button
          title="Mortgage Calculator"
          onPress={() =>
            this.props.navigator.showModal({
              screen: "example.mortgagecalculator", // unique ID registered with Navigation.registerScreen
              title: "Modal", // title of the screen as appears in the nav bar (optional)
              passProps: {}, // simple serializable object that will pass as props to the modal (optional)
              navigatorStyle: {}, // override the navigator style for the screen, see "Styling the navigator" below (optional)
              animationType: "slide-up" // 'none' / 'slide-up' , appear animation for the modal (optional, default 'slide-up')
            })
          }
          buttonStyle={{
            borderRadius: 18,
            backgroundColor: "#FF9212",
            marginHorizontal: 0,
            width: Dimensions.get("window").width * 0.5,
            alignSelf: "center",
            margin: 8,
            padding: 12
          }}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white"
  },
  homeList: {
    flex: 1,
    flexDirection: "row",
    width: Dimensions.get("window").width,
    height: 90,
    backgroundColor: "white",
    marginBottom: 6,
    shadowColor: "#000000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25
  },
  homeListThumbnail: {
    flex: 1
  },
  homeListContent: {
    flex: 2,
    justifyContent: "space-around",
    textAlign: "right",
    color: "white",
    padding: 10
  }
});
