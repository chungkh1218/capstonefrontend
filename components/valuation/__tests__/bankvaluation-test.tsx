import React from "react";
import BankValuation from "../bankvaluation";

import renderer from "react-test-renderer";

test("BankValuation renders correctly", () => {
  const tree = renderer.create(<BankValuation navigator />).toJSON();
  expect(tree).toMatchSnapshot();
});
