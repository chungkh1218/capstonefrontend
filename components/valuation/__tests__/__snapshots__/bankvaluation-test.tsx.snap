// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankValuation renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "name": "HSBC",
          "src": Object {
            "testUri": "../../../src/img/hsbc_logo.png",
          },
          "url": "https://www.hsbc.com.hk/personal/mortgages/property-valuation-tool.html",
        },
        Object {
          "name": "Hang Seng Bank",
          "src": Object {
            "testUri": "../../../src/img/hsb_logo.png",
          },
          "url": "https://www.hangseng.com/en-hk/e-valuation/address-search/#",
        },
        Object {
          "name": "DBS",
          "src": Object {
            "testUri": "../../../src/img/dbs_logo.png",
          },
          "url": "https://evalhk.dtz.com/e-valuation/DBSV2/Home/Index/en",
        },
        Object {
          "name": "Standard Chartered",
          "src": Object {
            "testUri": "../../../src/img/sc_logo.png",
          },
          "url": "https://www.sc.com/hk/borrow/mortgage-planner/property-valuation/",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "red",
                "flex": 1,
                "flexDirection": "row",
                "height": 90,
                "justifyContent": "space-between",
                "marginBottom": 6,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "padding": 20,
                }
              }
            >
              HSBC
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "justifyContent": "center",
                  "width": 187.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/img/hsbc_logo.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "red",
                "flex": 1,
                "flexDirection": "row",
                "height": 90,
                "justifyContent": "space-between",
                "marginBottom": 6,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "padding": 20,
                }
              }
            >
              Hang Seng Bank
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "justifyContent": "center",
                  "width": 187.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/img/hsb_logo.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "red",
                "flex": 1,
                "flexDirection": "row",
                "height": 90,
                "justifyContent": "space-between",
                "marginBottom": 6,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "padding": 20,
                }
              }
            >
              DBS
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "justifyContent": "center",
                  "width": 187.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/img/dbs_logo.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "red",
                "flex": 1,
                "flexDirection": "row",
                "height": 90,
                "justifyContent": "space-between",
                "marginBottom": 6,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "padding": 20,
                }
              }
            >
              Standard Chartered
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "justifyContent": "center",
                  "width": 187.5,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/img/sc_logo.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FF9212",
              "borderRadius": 18,
              "margin": 8,
              "marginHorizontal": 0,
              "padding": 12,
              "width": 375,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Mortgage Calculator
        </Text>
      </View>
    </View>
  </View>
</View>
`;
